llvm/lib/Target/RISCV/RISCVSubtarget.cpp

```diff
@@ -199,3 +200,25 @@ unsigned RISCVSubtarget::getMinimumJumpTableEntries() const {
              ? RISCVMinimumJumpTableEntries
              : TuneInfo->MinimumJumpTableEntries;
 }
+
+void RISCVSubtarget::overrideSchedPolicy(MachineSchedPolicy &Policy,
+                                         unsigned NumRegionInstrs) const {
+  // Do bidirectional scheduling since it provides a more balanced scheduling
+  // leading to better performance. This will increase compile time.
+  Policy.OnlyTopDown = false;
```

Comment:
IIRC setting OnlyBottomUp to false is sufficient to turn on bidirectional scheduler, since OnlyTopDown is default to false

Reply:
It does, but if you look at the other targets they use the same idiom and having both side by side is more readable (IMO).